src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2357,19 +2363,52 @@ private static NormalizedSnapshotSpanCollection GetSelectionSpans(ITextView text
                 return selection.SelectedSpans;
             }
 
+            /// <summary>
+            /// REPL clipboard format.
+            /// </summary>
+            private const string ReplFormat = "REPL";
+
             private DataObject Copy(NormalizedSnapshotSpanCollection spans)
             {
-                var text = spans.Aggregate(new StringBuilder(), GetTextWithoutPrompts, b => b.ToString());
+                var text = GetText(spans);
+                var blocks = GetTextBlocks(spans);
                 var rtf = _rtfBuilderService.GenerateRtf(spans, TextView);
                 var data = new DataObject();
                 data.SetData(DataFormats.StringFormat, text);
                 data.SetData(DataFormats.Text, text);
                 data.SetData(DataFormats.UnicodeText, text);
                 data.SetData(DataFormats.Rtf, rtf);
+                data.SetData(ReplFormat, blocks);
                 return data;
             }
 
-            private StringBuilder GetTextWithoutPrompts(StringBuilder builder, SnapshotSpan span)
+            /// <summary>
+            /// Get the text of the given spans as a simple concatenated string.
+            /// </summary>
+            private static string GetText(NormalizedSnapshotSpanCollection spans)
+            {
+                var builder = new StringBuilder();
+                foreach (var span in spans)
+                {
+                    builder.Append(span.GetText());
+                }
+                return builder.ToString();
+            }
+
+            /// <summary>
+            /// Get the text of the given spans as a serialized BufferBlock[].
+            /// </summary>
+            private string GetTextBlocks(NormalizedSnapshotSpanCollection spans)
+            {
+                var blocks = new List<BufferBlock>();
+                foreach (var span in spans)
+                {
+                    GetTextBlocks(blocks, span);
+                }
+                return BufferBlock.Serialize(blocks.ToArray());
+            }
+
+            private void GetTextBlocks(List<BufferBlock> blocks, SnapshotSpan span)
             {
                 // Find the range of source spans that cover the span.
                 var sourceSpans = GetSourceSpans(span.Snapshot);
```

Comment:
All spans will have the same snapshot, correct?  Would it be better to pass in the list of source spans?